src/main/scala/wdl4s/cwl/Implicits.scala

```diff
@@ -0,0 +1,98 @@
+package wdl4s.cwl
+
+import io.circe.Decoder
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.yaml.{parser => YamlParser}
+
+import io.circe.parser._
+import io.circe.shapes._
+import io.circe.generic.auto._
+import shapeless.{:+:, CNil, Coproduct}
+import cats.syntax.either._
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.string._
+import eu.timepit.refined._
+import io.circe.refined._
+import io.circe.literal._
+import cats.data.ValidatedNel
+import cats.data.Validated._
+import cats.syntax.traverse._
+import cats.instances.list._
+import cats.syntax.option._
+
+object Implicits {
+
+  implicit val cwlTypeDecoder = Decoder.enumDecoder(CwlType)
+  implicit val cwlVersionDecoder = Decoder.enumDecoder(CwlVersion)
+  implicit val scatterMethodDecoder = Decoder.enumDecoder(ScatterMethod)
+
+  type EVR = W.`"EnvVarRequirement"`.T => EnvVarRequirement
+  type IJR = W.`"InlineJavascriptRequirement"`.T => InlineJavascriptRequirement
+  type SR = W.`"SoftwareRequirement"`.T => SoftwareRequirement
+  type SWFR = W.`"SubworkflowFeatureRequirement"`.T => SubworkflowFeatureRequirement
+  type SDR = W.`"SchemaDefRequirement"`.T => SchemaDefRequirement
+  type DR = W.`"DockerRequirement"`.T => DockerRequirement
+  type IWDR = W.`"InitialWorkDirRequirement"`.T => InitialWorkDirRequirement
+  type SCR = W.`"ShellCommandRequirement"`.T => ShellCommandRequirement
+  type RR = W.`"ResourceRequirement"`.T => ResourceRequirement
+  type SFR = W.`"ScatterFeatureRequirement"`.T => ScatterFeatureRequirement
+  type MIFR = W.`"MultipleInputFeatureRequirement"`.T => MultipleInputFeatureRequirement
+  type SIER = W.`"StepInputExpressionRequirement"`.T => StepInputExpressionRequirement
+
+  type Target =
+    EVR :+:
+    IJR :+:
+    SR :+:
+    SWFR :+:
+    SDR :+:
+    DR :+:
+    IWDR :+:
+    SCR :+:
+    RR :+:
+    SFR :+:
+    MIFR :+:
+    SIER :+:
+    CNil
+
+  implicit val envvarD = Decoder[EVR]
+  implicit val ijr = Decoder[IJR]
+  implicit val sr = Decoder[SR]
+  implicit val sw = Decoder[SWFR]
+  implicit def sdr = Decoder[SDR]
+  implicit def rr = Decoder[RR]
+  implicit def iwdr = Decoder[IWDR]
+  implicit val dr = Decoder[DR]
+  implicit val scr = Decoder[SCR]
+  implicit val sfr = Decoder[SFR]
+  implicit val mifr = Decoder[MIFR]
+  implicit val sier = Decoder[SIER]
+
+  type V[A] = ValidatedNel[String,A]
+
+  def select[T](t: Target)(implicit selector: shapeless.ops.coproduct.Selector[Target,T]):ValidatedNel[String, T] =
+             t.select[T].toValidNel(s"Expecting a EnvVarRequirement but got $t instead.")
+
+  implicit val requirementArrayMapParser: Decoder[Array[Requirement]] =
```

Comment:
This is certainly not our worst offender out there but I wouldn't be sad if you did it :)